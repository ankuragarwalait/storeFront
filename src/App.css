/* Global Element Styles ------------------------------------------------------------------------------*/
body {
  margin: 0;
  padding: 0;
  font-family: sans-serif;
}

.App {
  text-align: center;
}

hr {
  margin: 1rem 0;
}

img.img-responsive {
  width: 100%;
  max-width: 100%;
}

.btn {
  display: block;
  padding: 1rem 3rem;
  font-size: 0.75rem; 
  color: white;
  border-width: 0;
  cursor: pointer;
  text-decoration: none;
}

.btn-primary {
  background-color: black;
}

.btn-secondary {
  background-color: grey;
}

.btn-tertiary {
  background-color: whote;
  color: black;
  border: 2px solid black;
}

.caret-link {
  cursor: pointer;
}

.caret-link::after {
  content: "";
  display: inline-block;
  margin-left: 0.2rem;
  vertical-align: middle;
  border-left: 0.2rem solid transparent;
  border-right: 0.2rem solid transparent;
  border-top: 0.2rem solid black;
}

/* Grid Styles ------------------------------------------------------------------------------ */

.container {
  max-width: 75rem;
  margin: 0 auto;
}

.row{
  display: flex;
  flex-wrap: wrap;
  padding: 0;
  justify-content: space-between;
}

.col-1 {
  flex: 1 0 8.33%;
}

.col-2 {
  flex: 1 0 16.66%;
}

.col-4 {
  flex: 1 0 30%;
  margin-right: 3.33%;
}

.col-6 {
  flex: 1 0 50%;
}

.col-8 {
  flex: 1 0 66.66%;
}

.col-4:nth-of-type(3n+3) {
  margin-right: 0;
}

/* Alignment Styles ------------------------------------------------------------------------------*/

.justify-center {
  justify-content: center;
}

.align-center {
  align-items: center;
}

.justify-around {
  justify-content: space-around;
}

.text-left {
  text-align: left;
}

.text-right {
  text-align: right;
}

/* Typography and Theme ------------------------------------------------------------------------------*/

.t-price, .t-brand {
  color: gray;
}

.t-brand {
  font-size: .75rem;
  font-weight: 500;
}

.t-font-xs {
  font-size: .25rem;
}

.t-font-s {
  font-size: .5rem;
}

.t-font-m {
  font-size: .75rem;
}

.t-cancel {
  float: right;
  cursor: pointer;
  color: gray;
  font-weight: bold;
}

.t-border-grey {
  border-color: grey;
}

.t-border-light-grey {
  border-color: lightgrey;
}

.t-color-black {
  color: black;
}

.t-color-white {
  color: white;
}

.t-color-grey {
  color: grey;
}

.t-color-lightgrey {
  color: lightgrey;
}
