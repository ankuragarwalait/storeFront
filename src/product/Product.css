.breadcrumb-container {
    margin-top: 2rem;
}

.breadcrumb-container * {
    font-size: .875rem;
    text-decoration: none;
    color: black;
}

.breadcrumb-prod-id {
    color: grey;
}

.prod-wrapper {
    margin-top: 4rem;
}

.prod-img {
    flex: 1 0 55%;
    height: 100%;
}

.prod-details-wrap {
    flex: 1 0 30%;
    font-family: auto;
    margin: 0 7.5%;
}

.t-name {
    font-size: 2.5rem;
    margin: .25rem 0;
    font-weight: normal;
}

.t-price {
    font-family: cursive;
    font-size: 1.125rem;
}

.t-prod-details {
    color: grey;
    font-size: 0.875rem;
}

.prod-wrapper .add-cart {
    padding-top: 1.25rem;
    padding-bottom: 1.25rem;
    margin-left: 0.5rem;
}