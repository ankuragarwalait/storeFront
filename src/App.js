import React, { Component } from 'react';
import { Route, Switch } from 'react-router-dom';
import './App.css';
import Category from "./category/Category";
import Cart from "./cart/Cart";
import Product from "./product/Product";
import Header from './shared/containers/Header';

class App extends Component {
  render() {
    return (
      <div className="App">    
        <Header cartCount={0} />
        <Switch>
          <Route exact path="/" component={Category} />
          <Route path="/cart" component={Cart}/>
          <Route path="/product/:id" component={Product}/>
        </Switch>
      </div>
    );
  }
}

export default App;
