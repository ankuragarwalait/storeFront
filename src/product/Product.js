import React, { Component } from 'react';
import './Product.css';
import BreadCrumb from '../shared/components/BreadCrumb';
import config from '../config';
import QtyWidget from '../shared/containers/QtyWidget';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { addToCart, updateCartQuantity } from '../store/actions/cartActions';


class Product extends Component {
    static propTypes = {
        product: PropTypes.shape({
            id: PropTypes.string,
            title: PropTypes.string,
            image: PropTypes.string,
            brand: PropTypes.string,
            description: PropTypes.string,
            price: PropTypes.number
        }),
        addToCart: PropTypes.func,
        updateQuantity: PropTypes.func
    }

    constructor(props) {
        super(props);
        this.addProdToCart = this.addProdToCart.bind(this);
        this.ref = React.createRef();
    }

    addProdToCart() {
        this.props.addToCart(Number(this.ref.current.value), this.props.product);
    }

    render() {
        const prodDetails = this.props.product;
        
        return (
            <div className="container">
                <BreadCrumb category="HOME" subCategory="PLATES" prodName={prodDetails.title} />
                <div className="row prod-wrapper">
                    <img className="img-responsive prod-img" src={`${config.imageURL}/${prodDetails.image}`} alt={prodDetails.id}/>
                    <div className="prod-details-wrap">
                        <p className="t-brand">{prodDetails.brand}</p>
                        <h1 className="t-name">{prodDetails.title}</h1>
                        <p className="t-price">{`$${parseFloat(prodDetails.price).toFixed(2)}`}</p>
                        <p className="t-prod-details">{prodDetails.description}</p>
                        <hr />
                        <div className="row justify-center align-center">
                            <QtyWidget qtyRef={this.ref} updateQuantity={this.props.updateQuantity} shouldAutoUpdate={false} prodId={prodDetails.id}/>
                            <button className="btn btn-primary add-cart" onClick={this.addProdToCart}>ADD TO CART</button>
                        </div>
                    </div>
                </div>
            </div>
        );
    }
}

const mapStateToProps = ({ products }, ownProps) => {
    return {
        product: products.find(product => product.id === ownProps.match.params.id)
    }
}

const mapDispatchToProps = (dispatch) => {
    return {
        addToCart: (qty, prodDetails) => dispatch(addToCart(qty, prodDetails)),
        updateQuantity: (qty, prodId) => dispatch(updateCartQuantity(qty, prodId)) 
    }
}

export default connect(mapStateToProps, mapDispatchToProps)(Product);
