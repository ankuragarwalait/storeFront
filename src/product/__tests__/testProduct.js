import renderer from 'react-test-renderer';
import React from 'react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import { mockStore } from "../../App.test";
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import {mount} from 'enzyme';
import Product from '../Product';

let tree = null;
let store = null;

describe('Test the product page', () => {
    beforeEach(()=> {
        configure({ adapter: new Adapter() });
        store = mockStore
    });

    it('test component matches snapshot', ()=> {
        const props = {
            match: {params: {id: "mashiko-yaki-saucer"}}
        }
        tree = renderer.create(<BrowserRouter>
            <Provider store={store}>
                <Product {...props}/>
            </Provider>
            </BrowserRouter>);
        const prodJson = tree.toJSON();
        expect(prodJson).toMatchSnapshot();
    })

    it('test add to Cart', ()=> {
        const props = {
            match: {params: {id: "mashiko-yaki-saucer"}}
        }

        tree = mount(<BrowserRouter>
            <Provider store={store}>
                <Product {...props}/>
            </Provider>
            </BrowserRouter>);
        tree.find('.add-cart').simulate('click');
        expect(store.getState().cart.itemCount).toBe(4);
    })
})