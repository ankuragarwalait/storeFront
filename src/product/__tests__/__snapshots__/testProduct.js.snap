// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the product page test component matches snapshot 1`] = `
<div
  className="container"
>
  <div
    className="breadcrumb-container"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      HOME
       / 
    </a>
    <a
      href="/"
      onClick={[Function]}
    >
      PLATES
       / 
    </a>
    <span
      className="breadcrumb-prod-id"
    >
      MASHIKO-YAKI SAUCER
    </span>
  </div>
  <div
    className="row prod-wrapper"
  >
    <img
      alt="mashiko-yaki-saucer"
      className="img-responsive prod-img"
      src="/media/mashiko-yaki-saucer.jpg"
    />
    <div
      className="prod-details-wrap"
    >
      <p
        className="t-brand"
      >
        Kiriko
      </p>
      <h1
        className="t-name"
      >
        Mashiko-Yaki Saucer
      </h1>
      <p
        className="t-price"
      >
        $10.00
      </p>
      <p
        className="t-prod-details"
      >
        Lorem ipsum dolor sit amet, consectetur adipiscing elit. Aliquam at purus pulvinar, placerat turpis ac, interdum metus. In eget massa sed enim hendrerit auctor a eget.
      </p>
      <hr />
      <div
        className="row justify-center align-center"
      >
        <div>
          <input
            className="counter-field"
            readOnly={true}
            value={1}
          />
          <div
            className="counter-control"
          >
            <button
              onClick={[Function]}
            >
              +
            </button>
            <button
              className="btn-dec"
              onClick={[Function]}
            >
              -
            </button>
          </div>
        </div>
        <button
          className="btn btn-primary add-cart"
          onClick={[Function]}
        >
          ADD TO CART
        </button>
      </div>
    </div>
  </div>
</div>
`;
